File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int right = numbers.size() - 1;
        
    for (int left = 0; left < numbers.size(); left++) {
        while (numbers[right] + numbers[left] > target) {
            right--;
        }
        if (numbers[right] + numbers[left] == target){
             return {left + 1, right + 1};
        }
    }
        
    return {};
        
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
       
        int right = numbers.length - 1;
        
        for (int left = 0; left < numbers.length; left++) {
            while (numbers[right] + numbers[left] > target) {
                right--;
            }
            if (numbers[right] + numbers[left] == target){
                int[] result = {left + 1, right + 1};
               return result;
            }
    }
        
    int[] result = {};
    return result;
        
    }
}